src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.WorkerThread.cs

```diff
@@ -71,6 +71,31 @@ private static void WorkerThreadStart()
                             CultureInfo.CurrentCulture = CultureInfo.InstalledUICulture;
                             CultureInfo.CurrentUICulture = CultureInfo.InstalledUICulture;
                         }
+                        else
+                        {
+                            // If we woke up but couldn't find a request, we need to update s_separated.counts that we are done working for now
```

Comment:
Another way may be to put this code at the end of the loop before looping around, and use continue for cases that can skip that work